packages/lesswrong/components/tagging/FilterMode.tsx

```diff
@@ -64,78 +92,92 @@ const styles = theme => ({
+  anchorEl?: any
 }) => {
-  const { LWTooltip } = Components
+  const { LWTooltip, PopperCard, TagPreview } = Components
   
-  const filterMoreNegative = () => {
-    if (mode === "Required") {
-      onChangeMode(maxFilterStrength);
-    } else if (mode === "Default") {
-      onChangeMode(-filterStrengthIncrement);
-    } else if (typeof mode === "number") {
-      if (mode<=-maxFilterStrength) onChangeMode("Hidden")
-      else if (mode-filterStrengthIncrement === 0) onChangeMode("Default");
-      else onChangeMode(mode-filterStrengthIncrement);
-    }
-  }
-  const filterMorePositive = () => {
-    if (mode === "Hidden") {
-      onChangeMode(-maxFilterStrength);
-    } else if (mode === "Default") {
-      onChangeMode(filterStrengthIncrement);
-    } else if (typeof mode === "number") {
-      if (mode>=maxFilterStrength) onChangeMode("Required");
-      else if (mode+filterStrengthIncrement === 0) onChangeMode("Default");
-      else onChangeMode(mode+filterStrengthIncrement);
-    }
-  }
-  
-  return <LWTooltip title={filterModeToTooltip(mode)} placement="bottom" className={classes.root}>
-    <span className={classes.label}>
-      {description}
-      {helpTooltip && <LWTooltip title={helpTooltip}>
-        <HelpIcon className={classes.helpIcon}/>
-      </LWTooltip>}
+  const { document: tag } = useSingle({
+    documentId: tagId,
+    collection: Tags,
+    fragmentName: "TagPreviewFragment",
+  })
+  return <span>
+    <span className={classes.tag}> 
+      {label} <span className={classes.filterScore}>
+        {filterModeToStr(mode)}
+      </span>
     </span>
-    
-    <span className={classNames(classes.button, {[classes.selected]: mode==="Hidden"})} onClick={ev => onChangeMode("Hidden")}>
-      Hidden
-    </span>
-    
-    <span className={classNames(classes.arrowButton, classes.arrowLeft, {[classes.selected]: mode<0})} onClick={filterMoreNegative}>
-      <UpArrowIcon/>
-    </span>
-    
-    <span className={classNames(classes.state)} onClick={ev => onChangeMode("Default")}>
-      {filterModeToStr(mode)}
-    </span>
-    
-    <span className={classNames(classes.arrowButton, classes.arrowRight, {[classes.selected]: mode>0})} onClick={filterMorePositive}>
-      <UpArrowIcon/>
-    </span>
-    
-    <span className={classNames(classes.button, {[classes.selected]: mode==="Required"})} onClick={ev => onChangeMode("Required")}>
-      Required
-    </span>
-  
-    {canRemove ? <div className={classes.closeButton} onClick={ev => {
-      if (onRemove)
-        onRemove();
-    }}>
-      X
-    </div> : <div className={classes.closeButton}/>}
-  </LWTooltip>
+    <PopperCard open={!!hover} anchorEl={anchorEl} placement="bottom" 
+      modifiers={{
+        flip: {
+          behavior: ["bottom-start", "top-end", "bottom-start"],
+          boundariesElement: 'viewport'
+        }
+      }}
+    >
+      <div className={classes.filtering}>
+        <span className={classes.filterLabel}>
+          Set Filter:
```

Comment:
Alas, having used the system, I actually think it's pretty important to give people granularity over how they want to set their filters. I am already sad because I can only give things +10 or +0, and would want to give the Rationality tag +7 for example.

Reply:
I think if we want to have more granularity it should just be an advanced setting that lets you input a number.

Reply:
I predict very few users would use more granularity tho.

Reply:
Actually I think what I'd like is to add the "Other" option, and then also remove two of the buttons, to avoid overwhelm. Maybe just "+10" and "+30".